static/js/components/email/EmailCompositionDialog.js

```diff
@@ -1,108 +1,99 @@
 // @flow
 import React from 'react';
 import Dialog from 'material-ui/Dialog';
-import R from 'ramda';
-import Grid, { Cell } from 'react-mdl/lib/Grid';
 
 import { FETCH_PROCESSING } from '../../actions';
 import { dialogActions } from '../inputs/util';
-import { getPropertyOrDefault } from '../../util/util';
-import { COURSE_EMAIL_TYPE, SEARCH_EMAIL_TYPE } from './constants';
-import type {
-  EmailState,
-  EmailValidationErrors
-} from '../../flow/emailTypes';
+import { isNilOrBlank } from '../../util/util';
+import type { EmailState } from '../../flow/emailTypes';
 
-const showValidationError = R.curry((getter, object: EmailValidationErrors) => {
-  let val = getter(object);
-  if (val !== undefined) {
-    return <span className="validation-error">{ val }</span>;
-  }
-});
-
-const showSubjectError = showValidationError(R.prop('subject'));
+export default class EmailCompositionDialog extends React.Component {
+  props: {
+    dialogVisibility:           boolean,
+    activeEmail:                EmailState,
+    title?:                     string,
+    subheadingRenderer?:        () => React$Element<*>,
+    closeAndClearEmailComposer: () => void,
+    closeEmailComposerAndSend:  () => void,
+    updateEmailFieldEdit:       () => void
+  };
 
-const showBodyError = showValidationError(R.prop('body'));
+  showValidationError = (fieldName: string): ?React$Element<*> => {
+    const { activeEmail: { validationErrors } } = this.props;
+    let val = validationErrors[fieldName];
+    if (val !== undefined) {
+      return <span className="validation-error">{ val }</span>;
+    }
+  };
 
-type EmailDialogProps = {
-  closeEmailDialog: () => void,
-  updateEmailEdit:  Function,
-  open:             boolean,
-  email:            EmailState,
-  sendEmail:        () => void,
-  title?:           string,
-  subheadingType?:  string
-};
+  renderDefaultSubheading = (subheading: ?string): React$Element<*> => (
+    <div className="subheading-section">
+      <h5 className="subheading">{ subheading }</h5>
+    </div>
+  );
 
-const renderSimpleSubheading = (subheading: string) => (
-  <div className="subheading-section">
-    <h5 className="subheading">{ subheading }</h5>
-  </div>
-);
+  renderSubheading = (): ?React$Element<*> => {
+    const {
+      activeEmail: { subheading },
+      subheadingRenderer
+    } = this.props;
+    if (isNilOrBlank(subheading)) {
+      return null;
+    } else {
+      let renderFunc = subheadingRenderer !== undefined ?
+        subheadingRenderer :
+        this.renderDefaultSubheading;
```

Comment:
👍  i think i wrote it like this b/c flow was being annoying, but i checked this again and it works